### PR TITLE
Better injection + add options in docker compose

### LINKED ISSUE #184: New environment variables not added to environment file
I have been running Lychee for a long time (probably about 4 to 5 years). I really struggled through this recent upgrade. Most of the problems I faced have already been discussed and / or resolved in other issues.

APP_FORCE_HTTPS (maybe APP_URL) needing to be set was my problem. Setting APP_FORCE_HTTPS=true in my docker compose file did not fix things (this resulted in me going down many rabbit holes related to other settings talked about in other issues - wasted many hours).

I forgot that there is an /conf/.env file in a volume. I didn't realize that the environment variables set in my compose file were being written to this file. But, that wasn't happening. Turns out `inject.sh` is only doing `sed`. APP_FORCE_HTTPS (and maybe APP_URL) were not being set in that file because they were never there.

I suggest for this project to do the following:
1. Check the result of `sed`, if it did not replace anything, append the new variable to the end of the file.
2. If not 1, print an error message if `sed` did not replace anything.

A bigger question: why do environment variables need to be in that file instead of being, you know, in the environment?

### PR DESCRIPTION
**Pull request by ildyria**
Fixes #184

### CODE REVIEW

File: inject.sh [hunk 1]
```diff
@@ -1,135 +1,140 @@
...
+ fi
 if [ "$DB_LOG_SQL_EXPLAIN" != '' ]; then
-    sed -i "s|DB_LOG_SQL_EXPLAIN=.*|DB_LOG_SQL_EXPLAIN=${DB_LOG_SQL_EXPLAIN}|i" /conf/.env
-fi
+    replace_or_insert "DB_LOG_SQL_EXPLAIN" "$DB_LOG_SQL_EXPLAIN"
+ fi
 if [ "$TIMEZONE" != '' ]; then
-    sed -i "s|TIMEZONE=.*|TIMEZONE=${TIMEZONE}|i" /conf/.env
-fi
+    replace_or_insert "TIMEZONE" "${TIMEZONE//\//\\\/}"
```
Comment (ildyria) [inline inject.sh:62]: `${TIMEZONE//\//\\\/}` instead of just `$TIMEZONE` is necessary to escape the `/` in the timezone like `Europe/Berlin`

Comment (d7415) [reply inject.sh:62]: That's why the sed was using `|` before.

File: inject.sh [hunk 2]
```diff
@@ -1,135 +1,140 @@
...
+# fi
+# if [ "$REDIS_PORT" != '' ]; then
+#     sed -i "s|REDIS_PORT=.*|REDIS_PORT=${REDIS_PORT}|i" /conf/.env
+# fi
+# if [ "$REDIS_PASSWORD" != '' ]; then
+#     sed -i "s|REDIS_PASSWORD=.*|REDIS_PASSWORD=${REDIS_PASSWORD}|i" /conf/.env
+# elif [ "$REDIS_PASSWORD_FILE" != '' ]; then
+#     value=$(<$REDIS_PASSWORD_FILE)
+#     sed -i "s|REDIS_PASSWORD=.*|REDIS_PASSWORD=${value}|i" /conf/.env
+# fi
```
Comment (d7415) [inline inject.sh:108]: I'm inclined to keep these while they still exist in Lychee's `.env.example`, but I could be convinced.

File: inject.sh [hunk 3]
```diff
@@ -1,135 +1,140 @@
 #!/bin/bash
+function replace_or_insert() {
+    # Voodoo magic: https://superuser.com/a/976712
+    grep -q "^${1}=" /conf/.env && sed "s/^${1}=.*/${1}=${2}/" -i /conf/.env || sed "$ a\\${1}=${2}" -i /conf/.env
```
Comment (d7415) [inline inject.sh:4]:
```suggestion
    grep -q "^${1}=" /conf/.env && sed -i "s|^${1}=.*|${1}=${2}|" /conf/.env || sed "$ a\\${1}=${2}" -i /conf/.env
```
That should fix the `TIMEZONE` issue

**Review by d7415 — COMMENTED** [review]
I don't have time to test right now, but looks good apart from that one possible bit of tidying by altering the `sed` command.

File: inject.sh (hunk 3 above)
Comment (ildyria) [reply inject.sh:4]: I will test that.

Comment (d7415) [reply inject.sh:4]: Tested with that and reverting the TIMEZONE change. I'll push to the branch then let you take a look and merge when you're happy.

**Review by d7415 — APPROVED** [review]
I'll let you looks at the `sed` (my test worked) and decide about the `REDIS_` items. Maybe we should drop them from the Lychee example and then both would match that way as I think you said Lychee doesn't benefit from it?

File: inject.sh (hunk 2 above)
Comment (ildyria) [reply inject.sh:108]: I would only put them once that they are actually used by Lychee. Not before. :)

**Comment by ildyria** [comment]
Tested LGTM.

**Pull request merged**